File: java/base/src/main/java/vsdk/toolkit/common/statistics/ReaderStatistics.java
package vsdk.toolkit.common.statistics;

public class ReaderStatistics {
    public int numberOfGeometries;
    public int numberOfCompounds;
    public int numberOfSurfaces;
    public int numberOfVertices;
    public int numberOfPatches;
    public int numberOfElements;
    public int numberOfLightSources;

    public ReaderStatistics() {
        numberOfGeometries = 0;
        numberOfCompounds = 0;
        numberOfSurfaces = 0;
        numberOfVertices = 0;
        numberOfPatches = 0;
        numberOfElements = 0;
        numberOfLightSources = 0;
    }

    public void reset() {
        numberOfGeometries = 0;
        numberOfCompounds = 0;
        numberOfSurfaces = 0;
        numberOfVertices = 0;
        numberOfPatches = 0;
        numberOfElements = 0;
        numberOfLightSources = 0;
    }
}

File: cpp/base/src/main/vsdk/toolkit/common/statistics/ReaderStatistics.cpp
#include "vsdk/toolkit/common/statistics/ReaderStatistics.h"

ReaderStatistics::ReaderStatistics():
    numberOfGeometries(0),
    numberOfCompounds(0),
    numberOfSurfaces(0),
    numberOfVertices(0),
    numberOfPatches(0),
    numberOfElements(0),
    numberOfLightSources(0)
{
}

void
ReaderStatistics::reset() {
    numberOfGeometries = 0;
    numberOfCompounds = 0;
    numberOfSurfaces = 0;
    numberOfVertices = 0;
    numberOfPatches = 0;
    numberOfElements = 0;
    numberOfLightSources = 0;
}
